relay-server/src/actors/upstream.rs

```diff
@@ -458,6 +483,53 @@ impl UpstreamRelay {
             .spawn(ctx);
     }
 
+    /// Adds a metric for the upstream request.
+    fn meter_result(
+        request: &UpstreamRequest,
+        send_result: &Result<ClientResponse, UpstreamRequestError>,
+    ) {
+        let status_code: StatusCode;
+        let mut sc = "-";
+        let result: &str;
+        match send_result {
+            Ok(ref client_response) => {
+                status_code = client_response.status();
+                sc = status_code.as_str();
+                result = "success";
+            }
+            Err(UpstreamRequestError::ResponseError(status_code, _)) => {
+                sc = status_code.as_str();
+                result = "response_error";
+            }
+            Err(UpstreamRequestError::PayloadFailed(_)) => result = "payload_failed",
+            Err(UpstreamRequestError::SendFailed(_)) => result = "send_failed",
+            Err(UpstreamRequestError::RateLimited(_)) => result = "rate_limited",
+            Err(UpstreamRequestError::NoCredentials)
+            | Err(UpstreamRequestError::ChannelClosed)
+            | Err(UpstreamRequestError::BuildFailed(_))
+            | Err(UpstreamRequestError::InvalidJson(_)) => {
+                // these are not errors caused when sending to upstream so we don't need to log anything
+                log::error!("meter_result called for unsupported error");
+                return;
```

Comment:
If there are no errors caused why not return silently?

Reply:
Because we should not end up here. These errors are generated outside the sending of the HTTP message, if we are called with one of these error it is a bug ( either I am wrong about the errors that we get in the current context or `meter_result` is called in the wrong place), any way I would like to know if the programming assumption is broken and not just silently ignore it.

Reply:
I see. In that case, let's also add `InvalidJson` to the supported errors, since theoretically the upstream could respond with an unparsable body. The other errors are fine here 👍

Reply:
We can add `InvalidJson` here but unfortunately we will never catch it here. We map json results to `InvalidJson` inside enqueue_query which is quite high in our stack ( we are mapping the error of the receiver oneshot channel).
The meter_result is applied much lower, to the future result handled by `handle_http_response` which is the `client_request.send()`.